### PR TITLE
JVM: Don't use deprecated Class#newInstance()

### PR DESCRIPTION
**Pull request by sideeffffect**
From https://github.com/portable-scala/portable-scala-reflect/pull/39#discussion_r604651184

### CODE REVIEW

File: jvm/src/main/scala/org/portablescala/reflect/InstantiatableClass.scala [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
+import java.lang.reflect.InvocationTargetException
+
+/**
+ * A wrapper for a class that can be instantiated.
  *
  *  @param runtimeClass
  *    The `java.lang.Class[_]` representing the class.
  */
 final class InstantiatableClass private[reflect] (val runtimeClass: Class[_]) {
+
```
Comment (gzm0) [inline jvm/src/main/scala/org/portablescala/reflect/InstantiatableClass.scala]: Spurious diff.

File: jvm/src/main/scala/org/portablescala/reflect/InstantiatableClass.scala [hunk 2]
```diff
@@ -1,26 +1,35 @@
 package org.portablescala.reflect
 
-/** A wrapper for a class that can be instantiated.
+import java.lang.reflect.InvocationTargetException
+
+/**
+ * A wrapper for a class that can be instantiated.
```
Comment (gzm0) [inline jvm/src/main/scala/org/portablescala/reflect/InstantiatableClass.scala]: Spurious diff.

File: jvm/src/main/scala/org/portablescala/reflect/InstantiatableClass.scala [hunk 3]
```diff
@@ -1,26 +1,35 @@
...
  *
  *  @param runtimeClass
  *    The `java.lang.Class[_]` representing the class.
  */
 final class InstantiatableClass private[reflect] (val runtimeClass: Class[_]) {
+
   /** A list of the public constructors declared in this class. */
   val declaredConstructors: List[InvokableConstructor] =
-    runtimeClass.getConstructors().map(new InvokableConstructor(_)).toList
+    runtimeClass.getConstructors.map(new InvokableConstructor(_)).toList
```
Comment (gzm0) [inline jvm/src/main/scala/org/portablescala/reflect/InstantiatableClass.scala]: Spurious diff.

File: jvm/src/main/scala/org/portablescala/reflect/InstantiatableClass.scala [hunk 4]
```diff
@@ -1,26 +1,35 @@
...
 
-  /** Instantiates this class using its zero-argument constructor.
+  /**
+   * Instantiates this class using its zero-argument constructor.
```
Comment (gzm0) [inline jvm/src/main/scala/org/portablescala/reflect/InstantiatableClass.scala]: Spurious diff.

**Review by gzm0 — CHANGES_REQUESTED** [review]
Only the diff issues

**Comment by sideeffffect** [comment]
@gzm0 those diffs are there so that it merges well with the original PR. It seemed easier this way. Or do you think it would be better to remove them anyway?

**Comment by gzm0** [comment]
> Or do you think it would be better to remove them anyway?

Yes I think removing them would be better. We should trade the other PR if there are any conflicts.

The reason for this is that we want a single PR to change one thing only (mich easier to understand the history that way).

**Comment by sideeffffect** [comment]
Sure thing! Done :heavy_check_mark:

**Comment by gzm0** [comment]
LGTM. We can merge this after you squash all the commits. (unless @sjrd has further comments).

**Comment by sideeffffect** [comment]
@gzm0 I would be more than happy to do the squashing myself, but if you don't mind me asking, why don't you just do the Squash-merge of this PR yourself? GitHub makes it a trivial one-click-button operation :smile_cat:

**Comment by gzm0** [comment]
Good question :) A bunch of reasons: On one hand habit from pre-GitHub support times. However, more importantly, Scala.js (and due to its history, this repo) uses `--no-ff` merges. So every merge creates an individual merge commit, even if the branch could be fast forwarded. Or in other words: The `master` branch only contains merge commits and special (release) commits directly. GitHub does not support this flavor of squash and merge.

Why do we do this: It allows us to have PRs that have multiple meaningful commits (which we use quite extensively in Scala.js core, for example: https://github.com/scala-js/scala-js/pull/4450).

**Comment by sideeffffect** [comment]
IMHO this PR is a good example where squash merge from GitHub UI would make things easier for everybody, but never mind, it's squashed now :wink:

**Review by sjrd — APPROVED** [review]

**Review by gzm0 — APPROVED** [review]

**Pull request merged**